File: App/Classes/Rim.m
#import "Rim.h"


@implementation Rim

@synthesize size, erd, offset, holeCount;


+ (NSDictionary *) dataMap {
    return [NSDictionary dictionaryWithObjectsAndKeys:
            [NSArray arrayWithObjects:@"pk", @"integer", nil], @"id",
            [NSArray arrayWithObjects:@"createdAt", @"datetime", nil], @"created_at",
            [NSArray arrayWithObjects:@"updatedAt", @"datetime", nil], @"updated_at",
            [NSArray arrayWithObjects:@"brand", @"string", nil], @"brand",
            [NSArray arrayWithObjects:@"description", @"string", nil], @"description",
            [NSArray arrayWithObjects:@"verified", @"boolean", nil], @"verified", 
            [NSArray arrayWithObjects:@"size", @"integer", nil], @"size",
            [NSArray arrayWithObjects:@"erd", @"float", nil], @"erd", 
            [NSArray arrayWithObjects:@"offset", @"float", nil], @"offset", 
            [NSArray arrayWithObjects:@"holeCount", @"integer", nil], @"hole_count", 
            nil];
}

-(NSString *)sizeDescription {
    NSDictionary *sizeDescriptions = [NSDictionary dictionaryWithObjectsAndKeys:
                                      @"19\"", [NSNumber numberWithInt:381],
                                      @"20\"", [NSNumber numberWithInt:406],
                                      @"20\"", [NSNumber numberWithInt:451],
                                      @"24\"", [NSNumber numberWithInt:507],
                                      @"24\"", [NSNumber numberWithInt:520],
                                      @"26\"", [NSNumber numberWithInt:559],
                                      @"650c", [NSNumber numberWithInt:571],
                                      @"650c", [NSNumber numberWithInt:590],
                                      @"700c", [NSNumber numberWithInt:622],
                                      @"27\"", [NSNumber numberWithInt:630],
                                      nil];
    return [sizeDescriptions objectForKey:[self size]];
}

- (void) dealloc {
    [size release];
    [erd release];
    [offset release];
    [holeCount release];
    [super dealloc];
}
@end

File: rails/parts/app/models/rim.rb
class Rim < ActiveRecord::Base

  validates_presence_of :brand, :description, :size, :erd, :offset, :hole_count


  def self.from_csv(row)
    rim = Rim.new :part_number => row[0],
                  :brand => row[2].strip,
                  :description => row[1].strip,
                  :size => row[3],
                  :erd => row[4].to_f,
                  :offset => row[5].to_f,
                  :hole_count => row[6],
                  :verified => true

    rim.description.gsub! /^#{rim.brand}\W+/i, ''
    if BRAND_ALIASES[rim.brand]
      rim.description.gsub! /^#{BRAND_ALIASES[rim.brand]}\W+/i, ''
    end
    rim
  end
  
end
